### DISCUSSION TITLE
Crypto token offline

### DISCUSSION THREAD

**Discussion by daud-silvergrey** [Keyfactor/signserver-ce#53] [Q&A]
I've installed EJBCA on my local machine using Docker and installed SignServer on one of my Ubuntu servers. I've created P12 and PEM certificates through EJBCA and installed the P12 certificate in my browser. SignServer CE is up and running. I've added a PDFSigner from the pdfsigner.properties template. All my workers are in OFFLINE mode. How can I sign a PDF? Please refer to the screenshots below for reference. Please help.

![Screen Shot 2023-08-24 at 5 48 00 PM](https://github.com/Keyfactor/signserver-ce/assets/142478009/67134035-3314-405a-918f-50e43cb8f19b)

![Screen Shot 2023-08-24 at 5 52 21 PM](https://github.com/Keyfactor/signserver-ce/assets/142478009/334d2e43-767c-4f81-aa47-21ccd829762b)
c3d1)

![Screen Shot 2023-08-24 at 5 50 21 PM](https://github.com/Keyfactor/signserver-ce/assets/142478009/c4049a94-4ac6-4c4e-9d55-ad48be43f098)

**Comment by primetomas** [comment]
Here are some good tutorial videos, showing the whole process, for some different signers and use cases.

https://www.youtube.com/watch?v=wMqPWKi3ukE
https://www.youtube.com/watch?v=sNIv0nYRs1g
https://www.youtube.com/watch?v=egG3Euw1GUk
https://www.youtube.com/watch?v=sfyJ3k-qYPY

**Reply by daud-silvergrey** [reply]
Thanks

**Reply by daud-silvergrey** [reply]
Hi,

Thank you. I've successfully created a crypto token and PDF signer following the video guide. My plan is to develop an app that allows users to perform document signing through my app's UI using SignServer's REST API. I've gone through the REST integration guide, but I'm still not entirely clear on some aspects. Please guide me.

I also have a few more questions:

Is it possible to use a single worker (let's say PDFSigner001) to sign a PDF file (e.g., contract1.pdf) by multiple users?
Can multiple users sign another PDF (e.g., contract2.pdf) through the same worker (e.g., PDFSigner001)?
Your assistance is greatly appreciated.

**Reply by mlundblad** [reply]
Yes, you could use a worker with multiple certs/keys depending the user.

There is the AliasSelector concept for doing this.

One example is using an AuthorizedUsernameAliasSelector that would derive the key alias to use from the authenticated user, using HTTP Basic auth.

**Reply by daud-silvergrey** [reply]
Hi,

Thank you for your response. Could you please suggest any tutorials or provide video links for REST integration?

Also, could you please let me know what I should use for `additionalProp1`, `additionalProp2`, `additionalProp3`, and `data` in the case of PdfSigner?

```json
{
  "metaData": {
    "additionalProp1": "value1",
    "additionalProp2": "value2",
    "additionalProp3": "value3"
  },
  "data": "dfgdf"
}
```

My signserver is running on 
https://sequidocs.sequifi.com/signserver

And when I am sending a POST Request to
https://sequidocs.sequifi.com/signserver/signserver/rest/v1/workers/1234/process

I am getting 405 error.
